.github/workflows/nestbuildmatrix.yml

```diff
@@ -2,135 +2,424 @@
+          make html |& tee sphinx-output.log
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
         with:
           path: |
-            /opt/hostedtoolcache/Python/**/site-packages/*
-            $HOME/.cache/pip
-          key: ${{ runner.os }}-python-${{ env.cache-name }}-${{ hashFiles('**/requirements.txt') }}
-          restore-keys: |
-            ${{ runner.os }}-python-${{ env.cache-name }}-
-            ${{ runner.os }}-python-
+            build/sphinx-output.log
+            build/userdoc/html/
 
-      - name: "Install Python dependencies"
+  sphinx-conda:
+    # as close as possible to the suggested user docs build in the documentation
+    runs-on: "ubuntu-20.04"
+    if: false
+    needs: [pydocstyle, rstcheck, vale]
+    steps:
+      - name: "Checkout repository content"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install Conda
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          auto-update-conda: true
+          python-version: "3.7"
+
+      - name: Install conda dependencies
+        shell: bash -l {0}
         run: |
-          python -m pip install --upgrade pip setuptools
-          # --force-reinstall required to ensure executable is linked from bin dir in PATH
-          python -m pip install --force-reinstall clang-format==13.0
-          python -c "import setuptools; print('package location:', setuptools.__file__)"
-          python -m pip install --force-reinstall --upgrade scipy 'junitparser>=2' numpy pytest pytest-timeout pytest-xdist mpi4py cython matplotlib terminaltables pandoc pandas 'pycodestyle>=2.9'
-          python -c "import pytest; print('package location:', pytest.__file__)"
-          pip list
+          conda info
+          conda env create -p conda
 
-      - name: "Find changed files"
+      - name: "Test-build documentation"
+        shell: bash -l {0}
         run: |
-          echo "CHANGED_FILES<<EOF" >> $GITHUB_ENV
-          echo "$(git diff --name-only ${{ github.event.pull_request.base.sha }}..${{ github.event.after }})" >> $GITHUB_ENV
-          echo 'EOF' >> $GITHUB_ENV
-          echo "GITHUB_ENV = $GITHUB_ENV"
-          cat $GITHUB_ENV
+          conda activate conda/
+          mkdir -pv build
+          cd build
+          cmake -DCMAKE_INSTALL_PREFIX=../install ..
+          make html |& tee sphinx-output.log
 
-      - name: "Static Code Analysis"
-        run: build_support/ci_build.sh
-        env:
-          xNEST_BUILD_TYPE: 'STATIC_CODE_ANALYSIS'
-          CHANGED_FILES: ${{ env.CHANGED_FILES }}
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          path: |
+            build/sphinx-output.log
+            build/userdoc/html/
 
-  test_linux:
-    if: ${{ !contains(github.event.head_commit.message, 'ci skip') }}
+  build_linux:
+    if: ${{ always() && !contains(github.event.head_commit.message, 'ci skip') }}
```

Comment:
Should `always()` be removed here so that this doesn't run if the previous steps fail?